src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -844,8 +983,8 @@ ss::future<cluster::init_tm_tx_reply> tx_gateway_frontend::init_tm_tx(
     auto delay_ms = _metadata_dissemination_retry_delay_ms;
     auto aborted = false;
 
-    auto tx_ntp_opt = _tx_coordinator_ntp_mapper.local().ntp_for(
-      kafka::group_id(tx_id));
+    auto tx_ntp_opt = co_await _tx_coordinator_ntp_mapper.local().ntp_for(
```

Comment:
`init_tm_tx` uses both static mapping (_tx_coordinator_ntp_mapper.local().ntp_for) and dynamic mapping (get_ntp) it should use only dynamic mapping like other rpc (add partition, add offset, end_txn)